### PR TITLE
Use testing library for change stream spec tests.

### PR DESCRIPTION
**Pull request by divjotarora**
GODRIVER-1266

Change-Id: Ice85b72144266e484ae35994538d873aac307087

### CODE REVIEW

File: mongo/integration/change_stream_spec_test.go [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+	Description      string                  `bson:"description"`
+	MinServerVersion string                  `bson:"minServerVersion"`
+	MaxServerVersion string                  `bson:"maxServerVersion"`
+	FailPoint        *mtest.FailPoint        `bson:"failPoint"`
+	Target           string                  `bson:"target"`
+	Topology         []mtest.TopologyKind    `bson:"topology"`
+	Pipeline         []bson.Raw              `bson:"changeStreamPipeline"`
+	Options          bson.Raw                `bson:"changeStreamOptions"`
+	Operations       []changeStreamOperation `bson:"operations"`
+	Exepctations     []*expectation          `bson:"expectations"`
```
Comment (iwysiu) [inline mongo/integration/change_stream_spec_test.go]: "Expectations"

Comment (divjotarora) [reply mongo/integration/change_stream_spec_test.go]: Good catch. Fixed.

File: mongo/integration/change_stream_spec_test.go [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+	"go.mongodb.org/mongo-driver/internal/testutil/assert"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/integration/mtest"
+)
+
+const (
+	changeStreamsTestsDir = "../../data/change-streams"
+)
+
+type changeStreamTestFile struct {
```
Comment (jyemin) [inline mongo/integration/change_stream_spec_test.go]: I like the removal of the abbreviations that are in the deleted test file!

File: mongo/integration/change_stream_spec_test.go [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+}
+
+func runChangeStreamTest(mt *mtest.T, test changeStreamTest, testFile changeStreamTestFile) {
+	mtOpts := mtest.NewOptions().MinServerVersion(test.MinServerVersion).MaxServerVersion(test.MaxServerVersion).
+		Topologies(test.Topology...).DatabaseName(testFile.DatabaseName).CollectionName(testFile.CollectionName)
+	mt.RunOpts(test.Description, mtOpts, func(mt *mtest.T) {
+		test.nsMap = make(map[string]struct{})
+
+		mt.ClearEvents()
+		var watcher watcher
```
Comment (jyemin) [inline mongo/integration/change_stream_spec_test.go]: Nice use of Go's duck typing

**Pull request merged**